### ISSUE TITLE
Plugins with dependancies still get loaded while the dependancies are missing.

### ISSUE THREAD

**Issue by ewanhowell5195** [JannisX11/blockbench#2290]
If you make a plugin that has dependancies, and those dependancies are not installed, blockbench will still try to load the plugin.

The expected behaviour is that the plugin will not load until all dependancies are met.

If this is intentional behaviour, maybe extend the dependancies system to have optional and required dependancies.

**Comment by JannisX11** [comment]
Yeah this wasn't part of the scope for the initial dependencies system, but it's definitely a good addition.

**Comment by ewanhowell5195** [comment]
i would consider this a bug however:

dependancies are not forced to load after the plugins they depend on
